go/libkb/leveldb.go

```diff
@@ -241,21 +242,25 @@ func (l *LevelDb) isCorrupt(err error) bool {
 	if strings.Contains(err.Error(), "corrupt") {
 		return true
 	}
-
+	// if our db is in a bad state with too many open files also nuke
+	if strings.Contains(strings.ToLower(err.Error()), "too many open files") {
+		return true
+	}
```

Comment:
I don't think running out of fds is all that uncommon. It would be nice to only do this if leveldb is actually the problem. Knowing how many fds leveldb tried to open would be ideal but that's probably annoying. Could we try to open some random other file, and only if that single open succeeds, return true?

Reply:
good idea (since this leveldb fd spike seems to be temporary, I guess you can't lsof it or anything)

Reply:
for simplicity i'd prefer to leave as is. loosing a db isn't the end of the world, if a user is really out of fds nuking the db might free it up and at least allow a log send which wasn't possible before